server.go

```diff
@@ -387,7 +388,8 @@ func newServer(listenAddrs []net.Addr, chanDB *channeldb.DB,
 		chansToRestore: chansToRestore,
 
 		invoices: invoices.NewRegistry(
-			chanDB, defaultFinalCltvRejectDelta,
+			chanDB, invoices.NewInvoiceExpiryWatcher(clock.DefaultClock{}),
```

Comment:
See comment above about the registry just initializing this itself given a closure for the current time. Given that, I don't think this struct needs to be private.

Reply:
Please see my above comment, but in general, dependencies should be injected rather privately constructed.

Reply:
I also have a slight preference for passing in just the `Clock`.